baxter_collaboration_lib/test/test_gripper.cpp

```diff
@@ -1,27 +1,232 @@
 #include <gtest/gtest.h>
-
 #include "robot_interface/gripper.h"
+#include <iostream>
 
 using namespace std;
 
-TEST(GripperTest, testConstructorDefaultValues)
+class gripperTester
 {
-    string    limb = "left";
-    bool use_robot =  false;
+    ros::NodeHandle             nh;
+    std::string               name;
+    std::string               limb;
+    ros::Publisher        prop_pub;
+    ros::Publisher       state_pub;
+    ros::Subscriber        cmd_sub; 
+    std::mutex    mutex_properties; 
```

Comment:
are you using this mutex? You should use a mutex to protect all the variables that can be accessed concurrently, ie in this case only cmd_value. I would change it to mutex_cmd , and use it whenever needed.